### ISSUE TITLE
Which part of the code handles 405 Method Not Allowed

### ISSUE THREAD

**Issue by siamackSeifi** [minio/minio#7173]
Hi.
I am trying to change the server code in a way that server could handle multiple domains for one instance (an array of `globalDomainName`)
For that I handled routers and methods in `cmd/api-router.go`
but I cant find which part of the code handles endpoint which when I enter `export MINIO_DOMAIN=mydomain.dev` handles it and allows `mybucket.mydomain.dev/` but it returns ` 405 Method Not Allowed.` for lets say `mybucket.myseconddomain.dev/`.

Is it possible for you to help me on this? or suggesting another approach which hadnt come to my mind?

tnx.

ps: I traced MINIO_DOMAIN and reached to globalDomainName variable and followed that but couldnt locate what I was looking for.

**Comment by kannappanr** [comment]
@siamackSeifi Looks like you want support for multiple domain names. We will send in a PR that has support for multiple domain names soon.

**Comment by siamackSeifi** [comment]
Exactly @kannappanr .
Thank you.

**Comment by siamackSeifi** [comment]
@kannappanr another matter is to add domain on the fly and not just on the startup.
Do you have that in mind for next release?
And for the sake of my understanding of how code works, could you please explain how is
THIS
> but I cant find which part of the code handles endpoint which when I enter `export MINIO_DOMAIN=mydomain.dev` handles it and allows `mybucket.mydomain.dev/` but it returns ` 405 Method Not Allowed.` for lets say `mybucket.myseconddomain.dev/`.

handled?
Where should I check in source code?

**Comment by harshavardhana** [comment]
Dynamically adding domains won't work @siamackSeifi as its not possible we need to initialize the HTTP server with the right muxers - you cannot dynamically change such handlers and its not allowed in HTTP. 

you should look at `cmd/api-router.go` for how domains are initialized.

**Comment by siamackSeifi** [comment]
Tnx @harshavardhana for the reply.
I checked `cmd/api-router.go`  as I said before.
I appended new domains to the routers like this:
```
routers = append(routers, apiRouter.Host("{bucket:.+}." + newDomain).Subrouter())
routers = append(routers, apiRouter.Host(newDomain + "/{bucket}").Subrouter())
```
and next in `for _, bucket := range routers {` loop handlers are getting assigned to routers
but even after that, I still get method not allowed.
So I thought there must be another piece of code to handle that other than this.
